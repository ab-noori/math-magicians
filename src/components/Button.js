import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ className, text }) => (
  <button type="button" className={`btn ${className}`}>
    {text}
  </button>
);

Button.propTypes = {
  className: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default Button;
