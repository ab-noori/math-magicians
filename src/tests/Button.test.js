import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Button className="my-class" text="Click me" onClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
