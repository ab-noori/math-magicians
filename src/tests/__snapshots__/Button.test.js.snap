// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="btn my-class"
  onClick={[Function]}
  type="button"
>
  Click me
</button>
`;
